src/App.tsx

```diff
@@ -3,58 +1,123 @@
+import cn from 'classnames';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
+import { useEffect, useState } from 'react';
 
+import { getUsers, getUserPosts } from './utils/fetchClient';
+import { User } from './types/User';
+import { Post } from './types/Post';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [currUser, setCurrUser] = useState<User | null>(null);
+  const [currPost, setCurrPost] = useState<Post | null>(null);
+  const [modalWindow, setModalWindow] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const searchPosts = (userId: number) => {
+    setLoading(true);
+    setError(null);
 
-              <Loader />
+    getUserPosts(userId)
+      .then(setPosts)
+      .catch(() => {
+        setError('Failed to load user posts. Please try again later.');
+      })
+      .finally(() => setLoading(false));
+  };
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  useEffect(() => {
+    setError(null);
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+    getUsers()
+      .then(setUsers)
+      .catch(() => {
+        setError('Failed to load users. Please try again later.');
+      });
```

Comment:
Similar to the previous suggestion, logging the error in the catch block can be beneficial for debugging and understanding what went wrong during the fetch operation.